custom/src/main/java/com/splunk/opentelemetry/TruncateCommandLineWhenMetricsEnabled.java

```diff
@@ -57,15 +62,66 @@ public Resource apply(Resource existing, ConfigProperties config) {
         return existing;
       }
 
-      String commandLine = existing.getAttribute(ResourceAttributes.PROCESS_COMMAND_LINE);
-      if (commandLine == null || commandLine.length() < 256) {
-        return existing;
+      Resource resource = existing;
+      if (resource.getAttribute(ResourceAttributes.PROCESS_COMMAND_ARGS) != null) {
+        List<String> newCommandArgs =
+            truncate(resource.getAttribute(ResourceAttributes.PROCESS_COMMAND_ARGS));
+        if (newCommandArgs != null) {
+          resource =
+              resource.merge(
+                  Resource.create(
+                      Attributes.of(ResourceAttributes.PROCESS_COMMAND_ARGS, newCommandArgs)));
+        }
+      }
+
+      String commandLine = resource.getAttribute(ResourceAttributes.PROCESS_COMMAND_LINE);
+      if (commandLine != null && commandLine.length() > MAX_LENGTH) {
+        String newCommandLine = commandLine.substring(0, MAX_LENGTH - 3) + "...";
+        resource =
+            resource.merge(
+                Resource.create(
+                    Attributes.of(ResourceAttributes.PROCESS_COMMAND_LINE, newCommandLine)));
+      }
+
+      if (existing != resource) {
+        logger.warning(
+            "Metrics are enabled. Truncating process.command_line and process.command_args resource attributes.");
+      }
+      return resource;
+    }
+
+    private static List<String> truncate(List<String> list) {
```

Comment:
So what's the reason behind it? The length of all values summed must be <255 as well?

Reply:
Can you add a comment saying that the JSON encoding of the list must be <255 chars long?

Reply:
done